src/status_im/chat/events.cljs

```diff
@@ -175,37 +154,36 @@
 (handlers/register-handler-fx
   :initialize-chats
   [(re-frame/inject-cofx :all-stored-chats)
+   (re-frame/inject-cofx :get-stored-messages)
    (re-frame/inject-cofx :stored-unviewed-messages)
-   (re-frame/inject-cofx :get-stored-unanswered-requests)
-   (re-frame/inject-cofx :get-last-stored-message)
-   (re-frame/inject-cofx :get-message-previews)]
+   (re-frame/inject-cofx :get-stored-unanswered-requests)]
   (fn [{:keys [db
                all-stored-chats
                stored-unanswered-requests
-               stored-unviewed-messages
-               get-last-stored-message
-               message-previews]} _]
-    (let [{:accounts/keys [account-creation?] :contacts/keys [contacts]} db
-          new-db (unviewed-messages-model/load-unviewed-messages db stored-unviewed-messages)
-          event  [:load-default-contacts!]]
+               get-stored-messages
+               stored-unviewed-messages]} _]
+    (let [{:accounts/keys [account-creation?] :contacts/keys [contacts]} db 
+          load-default-contacts-event [:load-default-contacts!]]
       (if account-creation?
-        {:db new-db
-         :dispatch event}
-        (let [chat->message-id->request (reduce (fn [acc {:keys [chat-id message-id] :as request}]
+        {:db db
+         :dispatch load-default-contacts-event}
+        (let [chat->unviewed-messages (unviewed-messages-model/index-unviewed-messages stored-unviewed-messages)
+              chat->message-id->request (reduce (fn [acc {:keys [chat-id message-id] :as request}]
                                                   (assoc-in acc [chat-id message-id] request))
                                                 {}
                                                 stored-unanswered-requests)
-              chats (->> all-stored-chats
-                         (map (fn [{:keys [chat-id] :as chat}]
-                                [chat-id (assoc chat
-                                                :last-message (get-last-stored-message chat-id)
-                                                :requests (get chat->message-id->request chat-id))]))
-                         (into {}))]
-          (-> new-db
-              (assoc-in [:message-data :preview] message-previews) 
+              chats (reduce (fn [acc {:keys [chat-id] :as chat}]
+                              (assoc acc chat-id
+                                     (assoc chat
+                                            :unviewed-messages (get chat->unviewed-messages chat-id)
+                                            :requests (get chat->message-id->request chat-id)
+                                            :messages (index-messages (get-stored-messages chat-id)))))
+                            {}
+                            all-stored-chats)]
+          (-> db 
               (assoc :chats chats)
               init-console-chat
```

Comment:
I don't think it is a good idea to have a function that takes a db and returns an effect map, especially when used in a threading macro. I was confused at first thinking you were wrongly trying to update the dispatch-n key in db until I realised init-console-chat was returing effects

Reply:
I fully agree and we should re-work it, the only reason why I'm not doing this is that console initialisation and functions around that are super messy and strange (partially stemming from the fact that console is dispatching messages while we don't have account db yet, so everytime console chat is opened, we actually have to dispatch them again...).
So I just carried on with it in this PR, we need another PR to fix the app and console init sequence properly.